### PR TITLE
PAINTROID-639 Make tool options disappear when tapping the tool

### PR DESCRIPTION
**Pull request by Lenkomotive**
[PAINTROID-639](https://jira.catrob.at/browse/PAINTROID-639)

## New Features and Enhancements
- added trigger to bottom nav bar to animate tooloptions
- added animation for tooloptions
- added option to assemble tooloptions according to tool (some of them will differ later)
- added tests for bottom nav bar

## Refactorings and Bug Fixes
- rename: BottomBrushToolOptions -> StrokeCapToolOption 
- rename: TopBrushToolOptions -> StrokeWidthToolOption
- fix: HandTool did show options too, now fixed

#### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: lib/ui/drawing_space/bottom_nav_bar.dart [hunk 1]
```diff
@@ -70,4 +74,25 @@ class BottomNavBar extends StatelessWidget {
...
+  void handleToolOptionsVisibility(WidgetRef ref) {
+    ref.read(toolOptionsProvider.notifier).toggleVisibility();
+  }
 }
+
+enum BottomNavBarItem {
+  TOOLS,
+  CURRENT_TOOL,
+  COLOR,
+  LAYERS,
```
Comment (bakicelebi) [inline lib/ui/drawing_space/bottom_nav_bar.dart]: 1 class/widget per file. I believe this is also true for enums.

File: lib/ui/drawing_space/bottom_nav_bar.dart [hunk 2]
```diff
@@ -70,4 +74,25 @@ class BottomNavBar extends StatelessWidget {
...
+    showModalBottomSheet(
+      context: context,
+      builder: (BuildContext context) => const SizedBox(
+        height: 270,
+        child: ToolsBottomSheet(),
+      ),
+    );
+  }
+
+  void handleToolOptionsVisibility(WidgetRef ref) {
```
Comment (bakicelebi) [inline lib/ui/drawing_space/bottom_nav_bar.dart:88]: kinda redundant function with 1 line.

File: lib/ui/drawing_space/tool_options.dart [hunk 3]
```diff
@@ -1,20 +1,59 @@
...
-        ],
+  Widget build(BuildContext context, WidgetRef ref) {
+    var options = getToolOptions(context, ref);
+    return  Padding(
+      padding: const EdgeInsets.all(8),
+      child: options,
+    );
+  }
+
+  Column getToolOptions(BuildContext context, WidgetRef ref) {
```
Comment (bakicelebi) [inline lib/ui/drawing_space/tool_options.dart]: Please make a widget out of these functions. https://medium.com/@s4.ali/flutter-code-review-dos-and-don-ts-and-best-practices-1-5d003035953e

File: test/widget/ui/bottom_control_navigation_bar_test.dart [hunk 4]
```diff
@@ -33,4 +37,192 @@ void main() {
         .selectTool(brushToolData)
         .then((_) => _.checkActiveToolIconAndLabel(brushToolData));
   });
+
+  testWidgets(
+      'test if stroke width tool option is invisible when starting app',
```
Comment (bakicelebi) [inline test/widget/ui/bottom_control_navigation_bar_test.dart]: There is lots of potential for code deduplication in here that can be added to bottom_nav_bar_interactions.dart

**Review by bakicelebi — CHANGES_REQUESTED** [review]

**Review by bakicelebi — APPROVED** [review]
👍🏻

File: packages/features/workspace_screen/lib/src/components/bottom_bar/bottom_nav_bar.dart [hunk 5]
```diff
@@ -5,32 +5,49 @@ import 'package:l10n/l10n.dart';
 import 'package:tools/tools.dart';
 import 'package:workspace_screen/workspace_screen.dart';
 
-class BottomNavBar extends StatelessWidget {
+class BottomNavBar extends ConsumerWidget {
```
Comment (juliajulie95) [inline packages/features/workspace_screen/lib/src/components/bottom_bar/bottom_nav_bar.dart:8]: I would not extend the ConsumerWidget
Usually you wrap the widget with a Consumer Widget and then you get a variable that is your instance of the reference you need. You can then hand this over to the function.

File: packages/features/workspace_screen/lib/src/components/bottom_bar/bottom_nav_bar.dart [hunk 6]
```diff
@@ -5,32 +5,49 @@ import 'package:l10n/l10n.dart';
...
+        height: 270,
+        child: ToolsBottomSheet(),
+      ),
+    );
+  }
+
+  void handleToolOptionsVisibility(WidgetRef ref) {
+    ref.read(toolOptionsVisibilityStateProvider.notifier).toggleVisibility();
+  }
+
```
Comment (juliajulie95) [inline packages/features/workspace_screen/lib/src/components/bottom_bar/bottom_nav_bar.dart]: Can you please put this under the override function?
overrides should always come first

**Review by juliajulie95 — APPROVED** [review]
Thank you! Good work :)

**Pull request merged**